build.jam

```diff
@@ -0,0 +1,55 @@
+# Copyright René Ferdinand Rivera Morell 2023-2024
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+
+require-b2 5.2 ;
+
+constant boost_dependencies :
+    /boost/assert//boost_assert
+    /boost/atomic//boost_atomic
+    /boost/config//boost_config
+    /boost/core//boost_core
+    /boost/date_time//boost_date_time
+    /boost/exception//boost_exception
+    /boost/filesystem//boost_filesystem
+    /boost/function_types//boost_function_types
+    /boost/fusion//boost_fusion
+    /boost/intrusive//boost_intrusive
+    /boost/io//boost_io
+    /boost/iterator//boost_iterator
+    /boost/move//boost_move
+    /boost/mpl//boost_mpl
+    /boost/optional//boost_optional
+    /boost/parameter//boost_parameter
+    /boost/phoenix//boost_phoenix
+    /boost/preprocessor//boost_preprocessor
+    /boost/property_tree//boost_property_tree
+    /boost/proto//boost_proto
+    /boost/range//boost_range
+    /boost/regex//boost_regex
+    /boost/smart_ptr//boost_smart_ptr
+    /boost/spirit//boost_spirit
+    /boost/system//boost_system
+    /boost/thread//boost_thread
+    /boost/throw_exception//boost_throw_exception
+    /boost/type_index//boost_type_index
+    /boost/type_traits//boost_type_traits
+    /boost/utility//boost_utility
+    /boost/winapi//boost_winapi
+    /boost/xpressive//boost_xpressive ;
```

Comment:
Boost.IO is not a public dependency.

Some of these dependencies are conditional and only present depending on the platform or build configuration.

And most importantly, the public dependencies need to be separate for `log` and `log_setup`. Preferably also separate `log` and `log_with_support`, as it is done in CMakeLists.txt.

Reply:
> I haven't dealt at all with optional dependencies in the modular work. As a main goal is for package management resolution. For which optional dependencies are considered a bad practice.

I'd say it's the opposite, as this is a way to reduce dependencies for users to satisfy.

Anyway, the dependencies are currently optional and separate for `log` and `log_setup`. I'd like it to remain this way after modularization.

There is no `log_with_support` in b2 currently because there was no point in it as the extra dependencies are header-only, and b2 didn't specify those before. You PR adds header-only dependencies, so it makes sense to introduce this target. Yes, this is new to b2, so I'm not strictly requiring it, but it would be nice to have.